crypto/x509v3/v3_genn.c

```diff
@@ -104,78 +105,54 @@ IMPLEMENT_ASN1_DUP_FUNCTION(GENERAL_NAME)
 
 static int edipartyname_cmp(const EDIPARTYNAME *a, const EDIPARTYNAME *b)
 {
-    int res = -1;
-
-    if ((a == NULL) || (b == NULL)) {
```

Comment:
Why are we deleting these checks?

Reply:
Additionally, the ASN.1 structure implies most fields cannot be NULL.
https://github.com/google/boringssl/commit/aa4ecb49269666c75919bc068028097c3b9cd42f

Reply:
In this asn1 mess I don't want to make all those assumptions. There is 0 downside to adding these checks.
What's the argument that these checks are not needed?

Reply:
> the ASN.1 structure implies most fields cannot be NULL.

Specifically, `d` type is `union`. It has been allocated automatically when the `GENERAL_NAME` is defined. So checking NULL is not necessary here.

```c
edipartyname_cmp(a->d.ediPartyName, b->d.ediPartyName);

union {
    ...
    EDIPARTYNAME *ediPartyName;
    ...
}  d;
```

Reply:
In general, for this kind of code that is rarely used, there is not really any reason to make these assumptions. Just add the NULL check, and you can avoid any potential issues in the future. You might also avoid having false-positives in static analysis tools.